ragna/_compat.py

```diff
@@ -53,9 +53,22 @@ def _anext() -> Callable[[AsyncIterator[T]], Awaitable[T]]:
     if sys.version_info[:2] >= (3, 10):
         anext = builtins.anext
     else:
-
-        async def anext(ait: AsyncIterator[T]) -> T:
-            return await ait.__anext__()
+        sentinel = object()
+
+        def anext(
+            ait: AsyncIterator[T],
+            default: T = sentinel,  # type: ignore[assignment]
+        ) -> Awaitable[T]:
+            if default is sentinel:
+                return ait.__anext__()
```

Comment:
I don't fully grok this. Do we not need to `await` this? And in what situation will this arise?

Reply:
> I don't fully grok this. Do we not need to `await` this?

We don't and we actually can't here. Note that `anext` is *not* an `async def` function. `anext` returns an awaitable, e.g.

```python
async_iterator = ...
awaitable = anext(async_iterator)
result = await awaitable
```

> And in what situation will this arise?

This is the default case, i.e. no `default` value is set. Let me refactor this function to make it more clear to what is going on.